lib/ddq-backend-mysql.js

```diff
@@ -0,0 +1,390 @@
+     * config's createMessageCycleLimit.
+     *
+     * @param {Object} ddqBackendInstance
+     * @param {Object} params
+     * @param {Function} callback
+     */
+    function trySendMessage(ddqBackendInstance, params, callback) {
+        if (cycleCounter < ddqBackendInstance.config.createMessageCycleLimit) {
+            cycleCounter += 1;
+            ddqBackendInstance.connection.query(`
+                INSERT INTO ?? (hash, message, topic)
+                VALUES (?, ?, ?)`,
+                [
+                    ddqBackendInstance.config.table,
+                    params.hash,
+                    params.message,
+                    params.topic
+                ], (err) => {
+                    if (err && err.code === "ER_DUP_ENTRY") {
+                        setRequeued(ddqBackendInstance, params);
+                    } else if (err) {
+                        ddqBackendInstance.emit(EMIT_ERR, err);
+                    } else {
+                        callback();
+                    }
+                }
+            );
+        } else {
+            callback(new Error("Could not send message."));
+        }
+    }
+
+
+    /**
+     * Updates a record. If there isn't an error, but no records are updated, it
+     * calls the trySendMessage method. Otherwise, it will call the callback
+     * that was provided to sendMessage.
+     *
+     * @param {Object} ddqBackendInstance
+     * @param {Object} params
+     * @param {Function} callback
+     */
+    setRequeued = function (ddqBackendInstance, params, callback) {
+        ddqBackendInstance.connection.query(`
+            UPDATE ??
+            SET requeued = true
+            WHERE hash = ?
+                AND isProcessing = true`,
+            [
+                ddqBackendInstance.config.table,
+                params.hash
+            ], (err, data) => {
+                if (err) {
+                    ddqBackendInstance.emit(EMIT_ERR, err);
+                } else if (data.affectedRows === 0) {
+                    trySendMessage(ddqBackendInstance, params);
+                } else {
+                    callback();
+                }
+            }
+        );
+    };
+
+
+    /**
+     * Class for the ddqBackendMySql. Inherits from EventEmitter.
+     */
+    class DdqBackendMySql extends EventEmitter {
+        /**
+         * Default DdqBackendMySql properties.
+         *
+         * @param {Object} backendConfig
+         */
+        constructor(backendConfig) {
+            configValidation.validateConfig(backendConfig);
+            super();
+            this.config = backendConfig;
+            this.connection = null;
+            this.currentlyPolling = false;
+            this.currentlyRestoring = false;
+            this.owner = crypto.randomBytes(64).toString("hex");
+            this.poller = null;
+            this.restorer = null;
+            this.topics = backendConfig.topics;
```

Comment:
Nothing validates this.  It could be a string, `null`, or an array.  Items in the array can be strings or `null`.